syft/pkg/cataloger/golang/test-fixtures/archs/Makefile

```diff
@@ -0,0 +1,8 @@
+GO_1_17 = /Users/jonas/.gvm/gos/go1.17.8/bin/go
+
+build:
+	GOARCH=arm64 GOOS=darwin $(GO_1_17) build -o hello-mach-o-arm64 main.go
```

Comment:
ideally we don't want to capture these in the git source tree, instead, keeping these in gitignored cache directories that CI can restore on an as needed basis. If the cache is not in place then the makefile during test setup could be invoked to populate the cache.